jaxlib/mosaic/gpu/custom_call.cc

```diff
@@ -119,7 +220,61 @@ void EnsureLLVMNVPTXTargetIsRegistered() {
   });
 }
 
-absl::StatusOr<std::pair<std::string, std::string>> GetSmAndPtxIsaVersion() {
+absl::StatusOr<int> GetLatestPtxasPtxIsaVersion() {
+  std::vector<const char*> ptxas_args = {"ptxas", "--input-as-string",
+                                         ".version 99.99", nullptr};
+  auto result = RunCUDATool("ptxas", ptxas_args);
+  if (result.ok()) {
+    return absl::InternalError("ptxas succeeded where it was expected to fail");
+  }
+  // Output message is of the form:
+  // ptxas application ptx input, line 1; fatal   : Unsupported .version 99.99; current version is '8.8'
+  std::vector<std::string> chunks =
+      absl::StrSplit(result.status().message(), '\'');
+  if (chunks.size() != 3) {
+    return absl::InternalError(
+        "Failed to locate PTX ISA version in ptxas error message");
+  }
+  std::vector<std::string> major_minor = absl::StrSplit(chunks[1], '.');
+  if (major_minor.size() != 2) {
+    return absl::InternalError(
+        absl::StrFormat("Expected PTX ISA version to be formatted as "
+                        "MAJOR.MINOR, instead got: %s",
+                        chunks[1]));
+  }
+  int major;
+  if (!absl::SimpleAtoi(major_minor[0], &major)) {
+    return absl::InternalError(
+        absl::StrFormat("Failed to parse PTX ISA major version, expected a "
+                        "parsable integer, instead got: %s",
+                        major_minor[0]));
+  }
+  int minor;
+  if (!absl::SimpleAtoi(major_minor[1], &minor)) {
+    return absl::InternalError(
+        absl::StrFormat("Failed to parse PTX ISA minor version, expected a "
+                        "parsable integer, instead got: %s",
+                        major_minor[1]));
+  }
+  return major * 10 + minor;
+}
+
+absl::StatusOr<std::string> GetPtxIsaVersion() {
+  int ptxas_latest_version;
+  TF_ASSIGN_OR_RETURN(ptxas_latest_version, GetLatestPtxasPtxIsaVersion());
```

Comment:
Nit: why not just put the declaration in here?

```
  TF_ASSIGN_OR_RETURN(int ptxas_latest_version, GetLatestPtxasPtxIsaVersion());
```

Reply:
Fixed, thank you.